### PR TITLE
closes #2224 re-vendor the latest Azure Storage SDK for better performance

### LINKED ISSUE #2224: re-vendor the latest Azure Storage SDK for better performance
Current version of Azure Storage SDK implements buildCanonicalizedHeader with unnecessary regex operation:
https://github.com/docker/distribution/blob/master/vendor/github.com/Azure/azure-sdk-for-go/storage/client.go#L269-L273

In our test it took about 9% CPU by itself, not to say GC pressure.
buildcanonicalizedstring-regex

TOP 10:
      flat  flat%   sum%        cum   cum%
    3660ms 10.54% 10.54%     6120ms 17.63%  runtime.scanobject
    1780ms  5.13% 15.67%     1780ms  5.13%  runtime.heapBitsForObject
    1760ms  5.07% 20.74%     1770ms  5.10%  runtime.greyobject
    1720ms  4.96% 25.70%     9960ms 28.69%  runtime.mallocgc
    1360ms  3.92% 29.62%     3560ms 10.26%  math/big.nat.divLarge
    1280ms  3.69% 33.30%     1280ms  3.69%  runtime.memclrNoHeapPointers
    1050ms  3.03% 36.33%     1140ms  3.28%  syscall.Syscall
     920ms  2.65% 38.98%      920ms  2.65%  runtime._ExternalCode
     690ms  1.99% 40.97%      690ms  1.99%  runtime.heapBitsSetType
     540ms  1.56% 42.52%      540ms  1.56%  math/big.addMulVVW

After change to plain string comparison the CPU usage was less than 2% by itself, 

buildcanonicalizedstring-string

      flat  flat%   sum%        cum   cum%
    1870ms  6.75%  6.75%     3150ms 11.38%  runtime.scanobject
    1650ms  5.96% 12.71%     1770ms  6.39%  syscall.Syscall
    1530ms  5.53% 18.24%     6720ms 24.27%  runtime.mallocgc
     970ms  3.50% 21.74%      970ms  3.50%  runtime._ExternalCode
     940ms  3.39% 25.14%      940ms  3.39%  runtime.greyobject
     810ms  2.93% 28.06%      810ms  2.93%  runtime.heapBitsSetType
     660ms  2.38% 30.44%      660ms  2.38%  runtime.heapBitsForObject
     640ms  2.31% 32.76%      640ms  2.31%  runtime.memclrNoHeapPointers
     630ms  2.28% 35.03%      630ms  2.28%  runtime.memmove
     480ms  1.73% 36.76%      480ms  1.73%  crypto/aes.decryptBlockGo

This should have been optimized in the latest Azure Storage SDK. Registry should consider to revendor to use the newer version.

### PR DESCRIPTION
**Pull request by yuwaMSFT2**
closes #2224 re-vendor the latest Azure Storage SDK for better performance

Signed-off-by: Yu Wang <yuwa@microsoft.com>

### CODE REVIEW

**Comment by yuwaMSFT2** [comment]
@stevvooe Could you please take a look? It's just re-vendor latest Azure GO SDK. Thanks!

**Comment by stevvooe** [comment]
LGTM

@dmcgowan PTAL

@dmp42 What do we need to do for testing here?

File: registry/storage/driver/azure/azure.go [hunk 1]
```diff
@@ -86,7 +86,8 @@ func New(accountName, accountKey, container, realm string) (*Driver, error) {
 	blobClient := api.GetBlobService()
 
 	// Create registry container
-	if _, err = blobClient.CreateContainerIfNotExists(container, azure.ContainerAccessTypePrivate); err != nil {
```
Comment (dmcgowan) [inline registry/storage/driver/azure/azure.go]: Why was this `azure.ContainerAccessTypePrivate` used before and no longer used? Is it deprecated now, is it the default, or was it just not needed.

**Comment by dmcgowan** [comment]
Looks like we currently don't have an Azure configuration for testing on our CI.
We would need...
`AZURE_STORAGE_ACCOUNT_NAME`
`AZURE_STORAGE_ACCOUNT_KEY`
`AZURE_STORAGE_CONTAINER`
`AZURE_STORAGE_REALM`

@dmp42 if we have an account somewhere to use I will update our CI

File: registry/storage/driver/azure/azure.go (hunk 1 above)
Comment (yuwaMSFT2) [inline registry/storage/driver/azure/azure.go]: It's the default. The header is optional when create container; without the header set, the container will be private. The new SDK doesn't provide the option to set the header. To change the permission, need to call Container.SetPermissions.

**Comment by dmcgowan** [comment]
LGTM

**Comment by yuwaMSFT2** [comment]
@stevvooe @dmcgowan @dmp42 
I ran the driver test manually. There was only one failure because that current Azure driver doesn't support PutContent of more than 64MB. This was an old change:
https://github.com/docker/distribution/pull/2088

Since it's not related to this re-vendor change, could you please help to merge?


_FAIL: /home/azureuser/oss/docker/src/github.com/docker/distribution/registry/storage/driver/testsuites/testsuites.go:852: DriverSuite.TestConcurrentStreamReads

/home/azureuser/oss/docker/src/github.com/docker/distribution/registry/storage/driver/testsuites/testsuites.go:866:
    c.Assert(err, check.IsNil)
... value driver.Error = driver.Error{DriverName:"azure", Enclosed:(*errors.errorString)(0xc42b09ecb0)} ("azure: uploading 134217728 bytes with PutContent is not supported; limit: 67108864 bytes")

OOPS: 33 passed, 1 FAILED
--- FAIL: Test (689.32s)_

**Pull request merged**